dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -746,6 +745,47 @@ QueryProcessingStage::Enum InterpreterSelectQuery::executeFetchColumns(Pipeline
                     "[InterpreterSelectQuery::executeFetchColumns] no region query",
                     ErrorCodes::LOGICAL_ERROR);
         }
+        else
+        {
+            // We may run query on a table other than manageable storage on mock test
+            if (auto managed_storage = std::dynamic_pointer_cast<IManageableStorage>(storage); managed_storage)
+            {
+                TableID table_id = managed_storage->getTableInfo().id;
+                // Only for (integration) test, because regions_query_info should never be empty if query is from TiDB or TiSpark.
+                // todo support partition table
+                auto & tmt = context.getTMTContext();
+                const auto regions = tmt.getRegionTable().getRegionsByTable(NullspaceID, table_id);
+                if (regions.empty())
+                {
+                    // We may run query on a table without any regions on mock test, keep going
+                    LOG_WARNING(
+                        log,
+                        "[InterpreterSelectQuery::executeFetchColumns] can not find any regions for the query, "
+                        "table_id={}",
+                        table_id);
+                }
+                else
+                {
+                    RUNTIME_CHECK_MSG(
+                        query_info.mvcc_query_info->regions_query_info.empty(),
+                        "the origin regions info is not empty! size={}",
+                        query_info.mvcc_query_info->regions_query_info.size());
+
+                    query_info.mvcc_query_info->regions_query_info.reserve(regions.size());
```

Comment:
Why not put this into a seperated function?

Reply:
Because these is just a stuff in a non-critical code path we will throw away sooner or later. And minimize the changes for reviewing.

Anyway, I've put the codes into separated functions

Reply:
Yes, that's what I mean. The next time when someone check these codes, it is easier to remove a function than a block of codes

Reply:
I means the whole `InterpreterSelectQuery` maybe thrown away someday ;)